src/Mvc/Mvc.Core/src/ApplicationModels/InferParameterBindingInfoConvention.cs

```diff
@@ -136,6 +136,25 @@ internal BindingSource InferBindingSourceForParameter(ParameterModel parameter)
         return BindingSource.Query;
     }
 
+    private bool IsService(Type type)
+    {
+        if (_serviceProviderIsService == null)
+        {
+            return false;
+        }
+
+        // IServiceProviderIsService will special case IEnumerable<> and always return true
+        // , so, in this case checking the element type instead
+        if (type.IsConstructedGenericType &&
+            type.GetGenericTypeDefinition() is Type genericDefinition &&
+            genericDefinition == typeof(IEnumerable<>))
```

Comment:
Are there any other types that could possible trigger false-positives from `IsService`? Looks like [IEnumerable](https://source.dot.net/#Microsoft.Extensions.DependencyInjection/ServiceLookup/CallSiteFactory.cs,555) might be the only one but just double-checking.

Reply:
Based on my research `IEnumerable` is the only one.